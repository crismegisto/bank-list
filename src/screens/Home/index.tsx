import React from 'react';
import { SafeAreaView, FlatList, ListRenderItem } from 'react-native';
import { useGetAllBanksQuery } from 'services/bank';
import { ListKeyExtractor } from 'interfaces/global';
import { Bank } from 'interfaces/bank';
import BankCard from 'components/BankCard';

function App() {
  const { data } = useGetAllBanksQuery(null);

  const renderItem: ListRenderItem<Bank> = ({ item }) => <BankCard {...item} />;

  const keyExtractor: ListKeyExtractor<Bank> = (_, index) => String(index);

  return (
    <SafeAreaView>
      <FlatList renderItem={renderItem} keyExtractor={keyExtractor} data={data} />
    </SafeAreaView>
  );
}

export default App;
