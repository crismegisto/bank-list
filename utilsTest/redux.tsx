import React from 'react';
import { ReactElement, ReactNode } from 'react';
import { Provider } from 'react-redux';
import { PreloadedState } from '@reduxjs/toolkit';
import { render, RenderOptions } from '@testing-library/react-native';
import { store as setupStore } from 'src/store';
import type { RootState, AppStore } from 'src/store';

export const initialStoreState = {};

interface reduxProps extends RenderOptions {
  preloadedState?: PreloadedState<RootState>;
  store?: AppStore;
}

export const renderWithProvider = (
  component: ReactElement,
  { preloadedState = {}, store = setupStore(preloadedState), ...renderOptions }: reduxProps = {}
) => {
  const wrapper = ({ children }: { children: ReactNode }): JSX.Element => {
    return <Provider store={store}>{children}</Provider>;
  };

  return { store, ...render(component, { wrapper, ...renderOptions }) };
};
