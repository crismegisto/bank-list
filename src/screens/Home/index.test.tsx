import React from 'react';
import { screen } from '@testing-library/react-native';

import { renderWithProvider } from '../../../utilsTest/redux';
import Home from './index';
import { mockBanks } from '../../../utilsTest/mockBanks';

jest.mock('services/bank', () => {
  const actualService = jest.requireActual('services/bank');
  return {
    ...actualService,
    useGetAllBanksQuery: jest.fn(() => ({
      data: mockBanks
    }))
  };
});

describe('Home Tests', () => {
  test('match snapshot', () => {
    renderWithProvider(<Home />);

    expect(screen.toJSON()).toMatchSnapshot();
  });

  test('Prove UI', async () => {
    renderWithProvider(<Home />);
    const title = await screen.findAllByTestId('title');
    expect(title).toHaveLength(5);
  });
});
