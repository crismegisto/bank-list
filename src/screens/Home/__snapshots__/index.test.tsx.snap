// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Tests match snapshot 1`] = `
<RCTSafeAreaView>
  <RCTScrollView
    data={
      [
        {
          "age": 10,
          "bankName": "Paga Todo",
          "description": "Banco Paga Todo es Para Todos",
          "url": "https://firebasestorage.googleapis.com/v0/b/stagingpagatodo-286214.appspot.com/o/Challenge%2Flogo-pagatodo.jpeg?alt=media&token=38b6ac4d-85ac-4288-bada-88eb5a0dec20",
        },
        {
          "age": 11,
          "bankName": "BBVA Bancomer",
          "description": "BBVA Bancomer Creando Oportunidades",
          "url": "https://firebasestorage.googleapis.com/v0/b/stagingpagatodo-286214.appspot.com/o/Challenge%2Flogo-bbva.jpeg?alt=media&token=435ed70e-061b-4a80-a216-c69cea04f068",
        },
        {
          "age": 9,
          "bankName": "Scotiabank México",
          "description": "Scotiabank Creando Tú decides, Nosotros te Asesoramos",
          "url": "https://firebasestorage.googleapis.com/v0/b/stagingpagatodo-286214.appspot.com/o/Challenge%2Flogo.scotiabank.jpg?alt=media&token=1029cc0b-7bff-4f5c-90f7-d96ca60f9619",
        },
        {
          "age": 7,
          "bankName": "Citibanamex",
          "description": "Citibanamex, lo mejor de México, lo mejor del mundo.",
          "url": "https://firebasestorage.googleapis.com/v0/b/stagingpagatodo-286214.appspot.com/o/Challenge%2Flogo-citi.png?alt=media&token=da97f3ad-34b2-4f7d-ae59-3169238993c7",
        },
        {
          "age": 5,
          "bankName": "Banregio",
          "description": "Banregio: Somos el banco de creadores",
          "url": "https://firebasestorage.googleapis.com/v0/b/stagingpagatodo-286214.appspot.com/o/Challenge%2Flogo-banregio.png?alt=media&token=ae605bda-5698-4bf8-9639-d4fdc9579b5c",
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={[]}
    viewabilityConfigCallbackPairs={[]}
  >
    <View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            {
              "backgroundColor": "white",
              "borderRadius": 6,
              "margin": 10,
              "padding": 10,
            }
          }
        >
          <View
            style={
              {
                "flexDirection": "row",
                "margin": 4,
              }
            }
          >
            <Image
              resizeMode="contain"
              source={
                {
                  "uri": "https://firebasestorage.googleapis.com/v0/b/stagingpagatodo-286214.appspot.com/o/Challenge%2Flogo-pagatodo.jpeg?alt=media&token=38b6ac4d-85ac-4288-bada-88eb5a0dec20",
                }
              }
              style={
                {
                  "height": 80,
                  "width": 120,
                }
              }
            />
            <View
              style={
                {
                  "justifyContent": "center",
                  "marginLeft": 18,
                }
              }
            >
              <Text
                style={
                  {
                    "fontSize": 18,
                    "fontWeight": "bold",
                  }
                }
                testID="title"
              >
                Paga Todo
              </Text>
              <Text>
                10
                 años
              </Text>
            </View>
          </View>
          <Text>
            Banco Paga Todo es Para Todos
          </Text>
        </View>
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            {
              "backgroundColor": "white",
              "borderRadius": 6,
              "margin": 10,
              "padding": 10,
            }
          }
        >
          <View
            style={
              {
                "flexDirection": "row",
                "margin": 4,
              }
            }
          >
            <Image
              resizeMode="contain"
              source={
                {
                  "uri": "https://firebasestorage.googleapis.com/v0/b/stagingpagatodo-286214.appspot.com/o/Challenge%2Flogo-bbva.jpeg?alt=media&token=435ed70e-061b-4a80-a216-c69cea04f068",
                }
              }
              style={
                {
                  "height": 80,
                  "width": 120,
                }
              }
            />
            <View
              style={
                {
                  "justifyContent": "center",
                  "marginLeft": 18,
                }
              }
            >
              <Text
                style={
                  {
                    "fontSize": 18,
                    "fontWeight": "bold",
                  }
                }
                testID="title"
              >
                BBVA Bancomer
              </Text>
              <Text>
                11
                 años
              </Text>
            </View>
          </View>
          <Text>
            BBVA Bancomer Creando Oportunidades
          </Text>
        </View>
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            {
              "backgroundColor": "white",
              "borderRadius": 6,
              "margin": 10,
              "padding": 10,
            }
          }
        >
          <View
            style={
              {
                "flexDirection": "row",
                "margin": 4,
              }
            }
          >
            <Image
              resizeMode="contain"
              source={
                {
                  "uri": "https://firebasestorage.googleapis.com/v0/b/stagingpagatodo-286214.appspot.com/o/Challenge%2Flogo.scotiabank.jpg?alt=media&token=1029cc0b-7bff-4f5c-90f7-d96ca60f9619",
                }
              }
              style={
                {
                  "height": 80,
                  "width": 120,
                }
              }
            />
            <View
              style={
                {
                  "justifyContent": "center",
                  "marginLeft": 18,
                }
              }
            >
              <Text
                style={
                  {
                    "fontSize": 18,
                    "fontWeight": "bold",
                  }
                }
                testID="title"
              >
                Scotiabank México
              </Text>
              <Text>
                9
                 años
              </Text>
            </View>
          </View>
          <Text>
            Scotiabank Creando Tú decides, Nosotros te Asesoramos
          </Text>
        </View>
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            {
              "backgroundColor": "white",
              "borderRadius": 6,
              "margin": 10,
              "padding": 10,
            }
          }
        >
          <View
            style={
              {
                "flexDirection": "row",
                "margin": 4,
              }
            }
          >
            <Image
              resizeMode="contain"
              source={
                {
                  "uri": "https://firebasestorage.googleapis.com/v0/b/stagingpagatodo-286214.appspot.com/o/Challenge%2Flogo-citi.png?alt=media&token=da97f3ad-34b2-4f7d-ae59-3169238993c7",
                }
              }
              style={
                {
                  "height": 80,
                  "width": 120,
                }
              }
            />
            <View
              style={
                {
                  "justifyContent": "center",
                  "marginLeft": 18,
                }
              }
            >
              <Text
                style={
                  {
                    "fontSize": 18,
                    "fontWeight": "bold",
                  }
                }
                testID="title"
              >
                Citibanamex
              </Text>
              <Text>
                7
                 años
              </Text>
            </View>
          </View>
          <Text>
            Citibanamex, lo mejor de México, lo mejor del mundo.
          </Text>
        </View>
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            {
              "backgroundColor": "white",
              "borderRadius": 6,
              "margin": 10,
              "padding": 10,
            }
          }
        >
          <View
            style={
              {
                "flexDirection": "row",
                "margin": 4,
              }
            }
          >
            <Image
              resizeMode="contain"
              source={
                {
                  "uri": "https://firebasestorage.googleapis.com/v0/b/stagingpagatodo-286214.appspot.com/o/Challenge%2Flogo-banregio.png?alt=media&token=ae605bda-5698-4bf8-9639-d4fdc9579b5c",
                }
              }
              style={
                {
                  "height": 80,
                  "width": 120,
                }
              }
            />
            <View
              style={
                {
                  "justifyContent": "center",
                  "marginLeft": 18,
                }
              }
            >
              <Text
                style={
                  {
                    "fontSize": 18,
                    "fontWeight": "bold",
                  }
                }
                testID="title"
              >
                Banregio
              </Text>
              <Text>
                5
                 años
              </Text>
            </View>
          </View>
          <Text>
            Banregio: Somos el banco de creadores
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
